project/webapp/templates/thread.html

```diff
@@ -90,4 +100,53 @@
     //
     // if (cw.ws.readyState == WebSocket.OPEN) cw.ws.onopen();
 </script>
+
+<script>
+    // Script for Share button dropdown menu.
+
+    // Allows dropdown menu to be closed by clicking away
+
+    $('html').click(function() {
+        document.getElementById('share-options').className = 'share-hidden';
+    });
+
+    $('#share-options').click(function(event){
+        event.stopPropagation();
+    });
+
+    $('#share-menu').click(function(event){
+        event.stopPropagation();
+    });
+
+    // Adds current thread URL and title to sharing link
+
+    links = document.getElementsByClassName('resp-sharing-button__link');
+    titleVars = ['text=', 'title=', 'caption=', 'subject='];
+    for (i = 0; i < links.length; i++) {
```

Comment:
Ok, this code is clever, but there is hopefully a simpler solution. I.e we have two nested loops, index checking, array indexing, nested property access, string slicing and concatenation. While this seems like it would work, it also may prove to be brittle and hard to debug.

Is there a way we can construct the `share_url` on the backend, with simple concatenation. Then pass it in to the `share_buttons` template as a context variable (for client rendering).

Reply:
If we can't find a 'cleaner', 'simpler' solution here, this one is alright. Instinct tells me there may be other approaches here.